src/sdk/pynni/nni/analysis_utils/sensitivity/torch/sensitivity_analysis.py

```diff
@@ -0,0 +1,257 @@
+                until all given sparsities are tested.
+
+        """
+        self.model = model
+        self.val_func = val_func
+        self.target_layer = OrderedDict()
+        self.ori_state_dict = copy.deepcopy(self.model.state_dict())
+        self.target_layer = {}
+        self.sensitivities = {}
+        if sparsities is not None:
+            self.sparsities = sorted(sparsities)
+        else:
+            self.sparsities = np.arange(0.1, 1.0, 0.1)
+        self.sparsities = [np.round(x, 2) for x in self.sparsities]
+        self.Pruner = L1FilterPruner
+        if prune_type == 'l2':
+            self.Pruner = L2FilterPruner
+        elif prune_type == 'fine-grained':
+            self.Pruner = LevelPruner
+        self.early_stop = early_stop
+        self.ori_acc = None  # original accuracy for the model
+        # already_pruned is for the iterative sensitivity analysis
+        # For example, sensitivity_pruner iteratively prune the target
+        # model according to the sensitivity. After each round of
+        # pruning, the sensitivity_pruner will test the new sensitivity
+        # for each layer
+        self.already_pruned = {}
+        self.model_parse()
+
+    @property
+    def layers_count(self):
+        return len(self.target_layer)
+
+    def model_parse(self):
+        for name, submodel in self.model.named_modules():
+            for op_type in SUPPORTED_OP_TYPE:
+                if isinstance(submodel, op_type):
+                    self.target_layer[name] = submodel
+                    self.already_pruned[name] = 0
+
+    def analysis(self, val_args=None, val_kwargs=None, start=0, end=None):
+        """
+        This function analyze the sensitivity to pruning for
+        each conv layer in the target model.
+        If %start and %end are not set, we analyze all the conv
+        layers by default. Users can specify several layers to
+        analyze or parallelize the analysis process easily through
+        the %start and %end parameter.
+
+        Parameters
+        ----------
+            start:
+                Layer index of the sensitivity analysis start
+            end:
+                Layer index of the sensitivity analysis end
+            val_args:
+                args for the val_function
+            val_kwargs:
+                kwargs for the val_funtion
+                The val_funtion will be called as:
+                    val_function(*val_args, **val_kwargs)
+
+        Returns
+        -------
+            sensitivities:
+                dict object that stores the trajectory of the
+                accuracy when the prune ratio changes
+        """
+        if not end:
+            end = self.layers_count
+        assert start >= 0 and end <= self.layers_count
+        assert start <= end
+        if val_args is None:
+            val_args = []
+        if val_kwargs is None:
+            val_kwargs = {}
```

Comment:
could directly assign `[]` and `{}` in input arguments.

Reply:
use empty list or dict as default values will trigger the pylint warning.
https://stackoverflow.com/questions/26320899/why-is-the-empty-dictionary-a-dangerous-default-value-in-python

Reply:
got it, thanks a lot!